src/main/java/mate/academy/dao/impl/BookDaoImpl.java

```
@@ -0,0 +1,119 @@
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(1, id);
+            try (ResultSet resultSet = statement.executeQuery()) {
+
+                if (resultSet.next()) {
+                    return Optional.of(parseBookFromResultSet(resultSet));
+                }
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get find book by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String sql = "SELECT * FROM books";
+        List<Book> books = new ArrayList<>();
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql);
+                ResultSet resultSet = statement.executeQuery()) {
+
+            while (resultSet.next()) {
+                books.add(parseBookFromResultSet(resultSet));
+            }
+            return books;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all books", e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        String sql = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            statement.executeUpdate();
+
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update book " + book, e);
```

This violates checklist item (implementation requirement):
"In update(Book): - Use PreparedStatement with SET of title and price by id. - Verify executeUpdate() > 0 to ensure update occurred. - Return the updated Book object (commonly same instance); handle cases where no row is updated by throwing DataProcessingException or by contract, but must not silently succeed."

Current code calls `statement.executeUpdate();` but ignores its return value and always returns the Book. If executeUpdate() returns 0 (no rows updated) the method will silently succeed. Change to inspect the returned int, throw DataProcessingException when it is 0 (with an informative message), and return the book only when rowsAffected > 0.